### PR TITLE
Feature/vs widget

### LINKED ISSUE #20: Feature/VS-widget
Ce widget représente l'affichage des 2 à 4 joueurs s'affrontant dans une partie.

image

image

image

### Il contient :
- [x] Une liste de pseudo pour les joueurs qui ont rejoint
- [x] Un bouton + pour inviter des joueurs lorsqu'il est dans l'état adéquat
- [x] Un status permettant de lui demander de seulement afficher ou bien de proposer aussi d'ajouter des joueurs

### PR DESCRIPTION
**Pull request by Laroche-Nicolas**

### CODE REVIEW

File: src/reusables/PlayersList.vue [hunk 1]
```diff
@@ -7,14 +7,51 @@
     <p class="players__vs">VS</p>
     <div class="players__list">
       <p>Joueur 3</p>
-      <p>+</p>
+      <p v-if="this.players.length === 4">Joueur 4</p>
```
Comment (MaxenceMottard) [inline src/reusables/PlayersList.vue]: no need for `this`

File: src/reusables/PlayersList.vue [hunk 2]
```diff
@@ -7,14 +7,51 @@
...
+      <button v-else @click="showModal" id="addPlayerButton" @keydown.esc="hideModal">+</button>
```
Comment (MaxenceMottard) [inline src/reusables/PlayersList.vue]: remove useless id

File: src/reusables/PlayersList.vue [hunk 3]
```diff
@@ -44,4 +89,68 @@ export default {
 .players p:nth-of-type(3) {
   text-align: right;
 }
+
+
+/* The Modal (background) */
```
Comment (MaxenceMottard) [inline src/reusables/PlayersList.vue]: Remove all comments please

File: src/reusables/PlayersList.vue [hunk 4]
```diff
@@ -44,4 +89,68 @@ export default {
...
+  top: 0;
+  width: 100%; /* Full width */
+  height: 100%; /* Full height */
+  overflow: auto; /* Enable scroll if needed */
+  background-color: rgb(0,0,0); /* Fallback color */
+  background-color: rgba(0,0,0,0.4); /* Black w/ opacity */
+}
+
+.shown {
+  display: block;
```
Comment (MaxenceMottard) [inline src/reusables/PlayersList.vue]: You can develop an animation. Example with simple opacity animation :

```css
.modal {
    pointer-events: none;
    cursor: none;
    opacity: 0;
    transition: 1s
}

.shown {
    pointer-events: auto;
    cursor: pointer;
    opacity: 1
}
```

File: src/reusables/PlayersList.vue [hunk 5]
```diff
@@ -44,4 +89,68 @@ export default {
...
+  left: 0;
+  top: 0;
+  width: 100%; /* Full width */
+  height: 100%; /* Full height */
+  overflow: auto; /* Enable scroll if needed */
+  background-color: rgb(0,0,0); /* Fallback color */
+  background-color: rgba(0,0,0,0.4); /* Black w/ opacity */
+}
+
+.shown {
```
Comment (MaxenceMottard) [inline src/reusables/PlayersList.vue]: It's better with BEM for this class name `modal--shown`

File: src/reusables/PlayersList.vue [hunk 6]
```diff
@@ -44,4 +89,68 @@ export default {
...
+}
+
+/* Modal Content/Box */
+.modal-content {
+  background-color: #fefefe;
+  margin: 15% auto; /* 15% from the top and centered */
+  padding: 20px;
+  border: 1px solid #888;
+  border-radius: 30px;
+  width: 40%; /* Could be more or less, depending on screen size */
```
Comment (MaxenceMottard) [inline src/reusables/PlayersList.vue]: I prefer use of `vw` instead of `%`here

File: src/reusables/PlayersList.vue [hunk 7]
```diff
@@ -44,4 +89,68 @@ export default {
...
+}
+
+.modal-content .modal__title {
+  font-size: 40px;
+  margin: 0 10px;
+  color: #616161;
```
Comment (MaxenceMottard) [inline src/reusables/PlayersList.vue]: Remove this color. We will standardize the colors later

**Review by MaxenceMottard — CHANGES_REQUESTED** [review]

**Review by zaktolba — APPROVED** [review]

**Review by MaxenceMottard — APPROVED** [review]

**Pull request merged**